tanner/api/api.py

```diff
@@ -1,128 +1,240 @@
-import json
+import datetime
 import logging
 import operator
-import aioredis
+from asyncio import TimeoutError
+from collections import ChainMap
+from json import dumps, loads, JSONEncoder
+from uuid import UUID
+
+import psycopg2
+from sqlalchemy import select, func
+
+from tanner.dbutils import COOKIES, OWNERS, PATHS, SESSIONS
+from tanner.utils.attack_type import AttackType
+
+
+class AlchemyEncoder(JSONEncoder):
+    def default(self, obj):
+        """JSON encoder function for SQLAlchemy special classes.
+            """
+        if isinstance(obj, datetime.date):
+            return obj.isoformat()
+        elif isinstance(obj, UUID):
+            return str(obj)
 
 
 class Api:
-    def __init__(self, redis_client):
-        self.logger = logging.getLogger('tanner.api.Api')
-        self.redis_client = redis_client
+    def __init__(self, pg_client):
+        self.logger = logging.getLogger("tanner.api.Api")
+        self.pg_client = pg_client
 
     async def return_snares(self):
+        """Returns a list of all the snares that are
+        connected to the tanner.
+
+        Returns:
+            [list] -- List containing UUID of all snares
+        """
         query_res = []
-        try:
-            query_res = await self.redis_client.smembers('snare_ids', encoding='utf-8')
-        except aioredis.ProtocolError as connection_error:
-            self.logger.exception('Can not connect to redis %s', connection_error)
-        return list(query_res)
+
+        async with self.pg_client.acquire() as conn:
+            stmt = select([SESSIONS.c.sensor_id], distinct=True)
+            rows = await (await conn.execute(stmt)).fetchall()
+            for r in rows:
+                query_res.append(str(r[0]))
+
+        return query_res
 
     async def return_snare_stats(self, snare_uuid):
+        """Returns the stats of the given snare
+
+        Arguments:
+            snare_uuid {uuid} -- UUID of snare
+
+        Returns:
+            [dict] -- Dictionary containing all stats snare.
+        """
         result = {}
-        result['total_sessions'] = 0
-        result['total_duration'] = 0
-        result['attack_frequency'] = {'sqli': 0,
-                                      'lfi': 0,
-                                      'xss': 0,
-                                      'rfi': 0,
-                                      'cmd_exec': 0}
-
-        sessions = await self.return_snare_info(snare_uuid)
-        if sessions == 'Invalid SNARE UUID':
-            return result
-
-        result['total_sessions'] = len(sessions)
-        for sess in sessions:
-            result['total_duration'] += sess['end_time'] - sess['start_time']
-            for attack in sess['attack_types']:
-                if attack in result['attack_frequency']:
-                    result['attack_frequency'][attack] += 1
+        result["total_sessions"] = 0
+        result["total_duration"] = 0
+        result["attack_frequency"] = {
```

Comment:
do we want to return only positive numbers or all zeros as well? if we want to return all the range, it should be more attacks

Reply:
yeah we can do that it returns all the attacks that are available, no matter what the type of attack is.